packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -660,24 +708,46 @@ const MetricForm: FC<MetricFormProps> = ({
               ) : (
                 <>
                   <div className="form-group">
-                    {table} Name
-                    <input
-                      type="text"
-                      required
-                      className="form-control"
-                      {...form.register("table")}
-                    />
+                    {supportsSchemaBrowser ? (
+                      <TypeaheadInput
+                        label={`${table} Name`}
+                        currentValue={form.watch("table")}
+                        onChange={(tableName, id) => {
+                          form.setValue("table", tableName);
+                          setTableId(id);
+                        }}
+                        options={tableOptions}
+                      />
+                    ) : (
+                      <Field
+                        label={`${table} Name`}
+                        {...form.register("table")}
+                        required
+                        type={"text"}
+                      />
+                    )}
                   </div>
                   {value.type !== "binomial" && (
                     <div className="form-group ">
-                      {supportsSQL ? "Column" : "Event Value"}
-                      <input
-                        type="text"
-                        required={value.type !== "count"}
-                        placeholder={supportsSQL ? "" : "1"}
-                        className="form-control"
-                        {...form.register("column")}
-                      />
+                      {supportsSchemaBrowser && tableId ? (
+                        <TypeaheadInput
+                          placeholder={column}
+                          label={supportsSQL ? "Column" : "Event Value"}
+                          options={columnOptions}
+                          currentValue={form.watch("column")}
+                          onChange={(columnName) =>
+                            form.setValue("column", columnName)
+                          }
+                        />
+                      ) : (
+                        <Field
+                          label={supportsSQL ? "Column" : "Event Value"}
+                          required={value.type !== "count"}
+                          placeholder={supportsSQL ? "" : "1"}
+                          className="form-control"
+                          {...form.register("column")}
+                        />
+                      )}
```

Comment:
This pattern seems like it would be reused a lot. Would it make sense to just always render the Type ahead input? If the list of options is empty, treat it just like a normal text input.

Reply:
@jdorn I went back and forth on this as I saw this pattern repeated a lot, but I did wonder if it made sense to render a regular input field inside of a component called `TypeaheadInput`. It makes things a bit cleaner, but I wonder if it could cause issues with the developer experience down the road? 

I've made these changes, but can revert if needed.